### ISSUE TITLE
onCellEditStop fires even when preProcessEditCellProps returns an error

### ISSUE THREAD

**Issue by BrianSHenderson** [mui/mui-x#5230]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When preProcessEditCellProps returns error: true onCellEditStop still fires, this would be okay if params included an attribute with whether there was a validation error from preProcessEditCellProps

This is with the newEditingApi

### Expected behavior 🤔

When validation fails don't fire onCellEditStop or include a parameter with whether validation has been successful.

### Steps to reproduce 🕹

Steps:

1. With preProcessEditCellProps: (params) => { return { ...params.props, error: true }
2. onCellEditStop={params => console.log(params)}

### Your environment 🌎

Using NextJS

### Order ID 💳 (optional)

#45829

**Comment by flaviendelangle** [comment]
@m4theushw can I let you this one ?

**Comment by BrianSHenderson** [comment]
Yes, I was trying to persist the data, I'll use processRowUpdate instead.

Thanks

**Comment by m4theushw** [comment]
I'll close this issue but feel free to open another one if you find any bug.
